src/components/certificates/CertificateForm.tsx

```diff
@@ -25,6 +31,13 @@ const NewCertificateForm: React.FC = () => {
 
   const [formState, setFormState] = useState<Certificate>(initialFormState);
```

Comment:
initial form data should actually be what you get from props. on Create new it will be this constant above, on edit it will be actuall data.